VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -274,4 +278,87 @@ void ViewItem::ensureColumnExists(int column)
 		nodes_.resize(column + 1);
 }
 
+//JSON storing/loading
+QJsonDocument ViewItem::toJson() const
+{
+	//Store all the nodes
+	QJsonArray nodes;
+	for (int col = 0; col < nodes_.size(); col++)
+		for (int row = 0; row < nodes_[col].size(); row++)
+			if (auto node = DCast<ViewItemNode>(nodes_[col][row]))
+				nodes.append(node->toJson(this));
+	//Store all the arrows
+	QJsonArray arrows;
+	for (auto key : arrows_.keys())
+		for (auto pair : arrows_[key])
+			arrows.append(arrowToJson(pair, key));
```

Comment:
Is there a way to make this method somehow belong to the arrow not write it down here in `ViewItem`?